### DISCUSSION TITLE
Adding Kryptor help wanted

### DISCUSSION THREAD

**Discussion by samuel-lucas6** [Homebrew/discussions#2795] [Writing Formulae/Casks] [answered]
I'd like to put my program called [Kryptor](https://github.com/samuel-lucas6/Kryptor) on Homebrew for macOS and Linux, but I'm rather overwhelmed by the documentation having never done anything like this before.

My desktop is running Windows, so I can't execute any of the commands. I did read about building from source rather than using a Cask when the program is open source and has a CLI, but using a Cask seems easiest and mirrors the Chocolatey package I've just tried to set up. Here's what I came up with based on examples:

```rb
cask "kryptor" do
  version "3.1.0"
  sha256 "dd3745f1b4925777af3e8d35b4187b16198ebbc0b3aa693e8f529c584c5de072"

  url "https://github.com/samuel-lucas6/Kryptor/releases/download/v#{version}/kryptor-macos.zip",
      verified: "github.com/samuel-lucas6/Kryptor/"
  name "Kryptor"
  desc "A simple, modern, and secure encryption and signing tool."
  homepage "https://www.kryptor.co.uk/"

  app "Kryptor.app"
end
```

However, this would only support macOS, not Linux. Should I not use a Cask? How can I support both macOS and Linux?

Many thanks.

**Comment by carlocab** [comment, marked as answer]
> How can I support both macOS and Linux?

The only way to do this is to build it from source in a formula, I'm afraid.

Submitting it as a cask is likely to lead to rejection: https://docs.brew.sh/Acceptable-Casks#rejected-casks Or, at the very least, you are probably going to be asked to submit it to Homebrew/core first. It will only be accepted into Homebrew/cask if it is not accepted into Homebrew/core.

**Reply by samuel-lucas6** [reply]
I'll have to try and find a C# example formula then.

**Reply by samuel-lucas6** [reply]
I'm not sure building from source will support the MacBook M1.
